src/main/webapp/app/shared/delete-dialog/delete-dialog.directive.ts

```diff
@@ -0,0 +1,64 @@
+import { DeleteDialogData, DeleteDialogService } from 'app/shared/delete-dialog/delete-dialog.service';
+import { Output, EventEmitter, Input, Directive, HostListener, Renderer2, ElementRef, OnInit } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+
+@Directive({ selector: '[jhiDeleteButton]' })
+export class DeleteDialogDirective implements OnInit {
+    @Input() entityTitle: string;
+    @Input() deleteQuestion: string;
+    @Input() deleteConfirmationText: string;
+    @Input() checkboxText: string;
+    @Input() additionalCheckboxText: string;
+    @Output() delete = new EventEmitter<any>();
+
+    deleteTextSpan: HTMLElement;
+
+    constructor(private deleteDialogService: DeleteDialogService, private renderer: Renderer2, private el: ElementRef, private translateService: TranslateService) {}
+
+    /**
+     * This method appends classes and type property to the button on which directive was used, additionally adds a span tag with delete text.
+     * We can't use component, as Angular would wrap it in it's own tag and this will break button grouping that we are using for other buttons.
+     */
+    ngOnInit() {
+        // set button classes and submit property
+        this.renderer.addClass(this.el.nativeElement, 'btn');
+        this.renderer.addClass(this.el.nativeElement, 'btn-danger');
+        this.renderer.addClass(this.el.nativeElement, 'btn-sm');
+        this.renderer.addClass(this.el.nativeElement, 'mr-1');
```

Comment:
It might be possible to do this with [`@HostBinding()`](https://www.thecodecampus.de/blog/angular-2-use-hostbindings-set-class/), but I am not sure if this would be better.
So I just leave this here for reference, but it can stay the way it is.

Reply:
Yes it is possible. I thought it is cleaner though to have everything in one place and not combine `@HostBinding` with `Renderer2` logic